### ISSUE TITLE
Check that tutorials run as notebooks

### ISSUE THREAD

**Issue by maxnoe** [gammapy/gammapy#4696]
**Is your feature request related to a problem? Please describe.**

It seems the tutorials are not tested as notebooks anymore since the conversion to jupytext.

Executing the event_sampling tutorial I got an error "Did not find line magic function %%time", 
which seems to come from a missing cell boundary (%%time in the middle of a cell instead of at the beginning as first line).

**Describe the solution you'd like**

Check that tutorials work when run as notebook.

**Comment by Astro-Kirsty** [comment]
Thanks @maxnoe. I executed the `event_sampling` tutorial on my local computer and I have no errors. I tried with both the dev version and v1.1 

Could you explain further where this error is originating from?

**Comment by adonath** [comment]
The tutorials are executed as notebooks as part of the docs build. I don't think I saw that error before.

**Comment by maxnoe** [comment]
On the current main branch, in the development environment, I do:

```
jupyter notebook examples/tutorials/analysis-3d/event_sampling.py
```

This results in a cell under the section of  
> Extended source using a template

that contains a `%%time` command in the middle of the cell:

![wrong_timeit](https://github.com/gammapy/gammapy/assets/5488440/9d7645fc-1d79-4923-a3ab-bfd203930811)

**Comment by maxnoe** [comment]
It happens a couple more times further down in the notebook.

**Comment by Astro-Kirsty** [comment]
`jupyter notebook examples/tutorials/analysis-3d/event_sampling.py`

This line will not work? Because you are trying to execute a .py script with jupyter? 

Maybe I misunderstood what you mean?

**Comment by AtreyeeS** [comment]
I suppose this is resolved now on our latest dev? Can we clopse this issue @Astro-Kirsty @maxnoe ?

**Comment by maxnoe** [comment]
No, this should still be an issue. I boils down to an incompatibility between sphinx-gallery (what is used to Redner the scripts in the docs and export them back to notebooks) and jupytext (which is what allows you to open some .py files directly in Jupiter).


See: https://github.com/mwouts/jupytext/issues/1169

So it's currently just not possible to work with the tutorials locally using the jupyter interface

**Comment by AtreyeeS** [comment]
Thanks @maxnoe ! So the issue should be resolved withing jupyter?

>So it's currently just not possible to work with the tutorials locally using the jupyter interface

Opening `jupyter lab/notebook` and then launching the tutorials work fine on my machine. I see it automatically broken into the relevant blocks, and the `%% time` command works fine as well.

Shall we close the issue here?

**Comment by maxnoe** [comment]
> Opening jupyter lab/notebook and then launching the tutorials work fine on my machine.

Which tutorials? The notebook downloaded from the docs page? Or the local .py file?

**Comment by AtreyeeS** [comment]
> > Opening jupyter lab/notebook and then launching the tutorials work fine on my machine.
> 
> Which tutorials? The notebook downloaded from the docs page? Or the local .py file?

The local .py files in my gammapy repo

**Comment by registerrier** [comment]
In  practice, sphinx_gallery seems to strip the `%%time` magic commands. They don't appear neither in the gallery nor in the notebooks produced. 

This means we can probably remove them from the .py files altogether. 

The only thing failing is when one runs something like `jupytext --execute some_tutorial.py` or try to run the notebook produced with `jupytext --to ipynb some_tutorial.py`.

We could add a scheduled action running `jupytext --execute examples/tutorials/*/*.py` to make sure this runs fine, if this is important. Otherwise remove the `%%time` from the python scripts.

**Comment by Astro-Kirsty** [comment]
I would suggest we remove the `%%time` as it is not beneficial if it doesn't show in the tutorials. Will make a PR for that.
